scripts/lib/LinkChecker.ts

```diff
@@ -0,0 +1,128 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2023.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import path from "node:path";
+
+const DOCS_ROOT = "./docs";
+const CONTENT_FILE_EXTENSIONS = [".md", ".mdx", ".ipynb"];
+
+export class File {
+  readonly path: string;
+  anchors: string[];
+
+  constructor(path: string) {
+    this.path = path;
+    this.anchors = [];
+  }
+
+  setAnchors(anchors: string[]) {
+    this.anchors = anchors;
+  }
+}
+
+export class Link {
+  readonly value: string;
+  readonly anchor: string;
+  readonly origins: string[];
+  readonly isExternal: boolean;
+
+  constructor(linkString: string, origins: string[]) {
+    /*
+     * linkString: Link as it appears in source file
+     *    origins: Paths to source file containing link
+     */
+
+    const splitLink = linkString.split("#", 2);
+    this.value = splitLink[0];
+    this.anchor = splitLink.length > 1 ? `#${splitLink[1]}` : "";
+    this.origins = origins;
+    this.isExternal = linkString.startsWith("http");
+  }
+
+  resolve(origin: string): string[] {
+    /*
+     * Return list of possible paths link could resolve to
+     */
```

Comment:
Reminder to move this docstring and all others to live above the method, and start with `/**` on the first line. (JSDoc)